### ISSUE TITLE
TypeNamesShouldNotMatchNamespaces (CA1724) reports on WinForms designer-generated code

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#949]
#### Analyzer package
Microsoft.ApiDesignGuidelines.Analyzers

#### Analyzer
CA1724: TypeNamesShouldNotMatchNamespaces

#### Repro steps
1. Create a WinForms application
2. Open Project properties > Resources and add a string resource
3. `install-package Microsoft.ApiDesignGuidelines.Analyzers -pre`
4. CA1724 reported on generated code in `Resources.Designer.cs`:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1724	The type name Resources conflicts in whole or in part with the namespace name 'System.Resources' defined in the .NET Framework. Rename the type to eliminate the conflict.	WindowsFormsApplication1	e:\bart\documents\visual studio 2015\Projects\WindowsFormsApplication1\WindowsFormsApplication1\Properties\Resources.Designer.cs	25	Active
```

#### Expected behavior
No reporting on designer-generated code.

#### Actual behavior
Above repro reports CA1724.

**Comment by mavasani** [comment]
Should get fixed with https://github.com/dotnet/roslyn-analyzers/pull/964

**Comment by bkoelman** [comment]
This still repros. It reports on source location: `E:\Bart\Documents\Visual Studio 2015\Projects\WindowsFormsApplication1\WindowsFormsApplication1\Properties\Resources.Designer.cs`

Stepping through the analyzer, I see https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Core/TypeNamesShouldNotMatchNamespaces.cs#L54 get hit, which should suppress reporting in generated code. 

But I'm still hitting the breakpoint at https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.ApiDesignGuidelines.Analyzers/Core/TypeNamesShouldNotMatchNamespaces.cs#L96.

![image](https://cloud.githubusercontent.com/assets/10324372/15130342/ef84e240-1647-11e6-9129-08effa70db8a.png)

**Comment by mavasani** [comment]
Thanks @bkoelman. Should be fixed with #967
